### PR TITLE
Enable code/dependency scanning

### PR DESCRIPTION
**Pull request by omrishiv**
I addressed all the security issues in the dependencies in #294 so this is currently not reporting anything. I guess we wait until the next one?

### CODE REVIEW

**Comment by omrishiv** [comment]
Fixes #135

File: .github/workflows/code-scanner.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      - run: ls -al
+      - name: Scan current project
+        id: scan
+        uses: anchore/scan-action@v3
+        with:
+          fail-build: false
+          path: "."
+
+      - name: Upload scan results to GitHub Security tab
+        uses: github/codeql-action/upload-sarif@v2
```
Comment (nabuskey) [inline .github/workflows/code-scanner.yaml:29]: This is deprecated. We need to update. https://github.com/cnoe-io/idpbuilder/actions/runs/9605967327/job/26494616601?pr=323

Comment (omrishiv) [reply .github/workflows/code-scanner.yaml]: ✅

**Review by nabuskey — APPROVED** [review]
LGTM

**Pull request merged**